gateway/src/main/kotlin/Event.kt

```diff
@@ -734,19 +738,31 @@ public data class GuildScheduledEventUpdate(val event: DiscordGuildScheduledEven
 public data class GuildScheduledEventDelete(val event: DiscordGuildScheduledEvent, override val sequence: Int?) :
     DispatchEvent()
 
-public data class GuildScheduledEventUserAdd(
-    val eventId: Snowflake,
-    val userId: Snowflake,
-    val guildId: Snowflake,
-    override val sequence: Int?,
-) : DispatchEvent()
+public data class GuildScheduledEventUserAdd(val data: Data, override val sequence: Int?) : DispatchEvent() {
+
+    @Serializable
+    public data class Data(
+        @SerialName("guild_scheduled_event_id")
+        val guildScheduledEventId: Snowflake,
+        @SerialName("user_id")
+        val userId: Snowflake,
+        @SerialName("guild_id")
+        val guildId: Snowflake,
+    )
+}
```

Comment:
I'm not sure about this nested `data class Data`, it's duplicated in `GuildScheduledEventUserRemove`, can someone think of a better name to share this between both?

Reply:
sounds like a ScheduledEventMetaData or (data)?

Reply:
renamed it to `GuildScheduledEventUserMetadata`, it's also shared now